odoo/sql_db.py

```diff
@@ -242,18 +243,11 @@ def dictfetchone(self) -> dict[str, typing.Any] | None:
         raise NotImplementedError
 
     def dictfetchmany(self, size: int) -> list[dict[str, typing.Any]]:
-        res: list[dict[str, typing.Any]] = []
```

Comment:
Does the BaseCursor abstraction is still needed at all ?

Reply:
Ho I see, I forgot that TestCursor was move. Then the abstraction still make sense.

Then you can just `raise NotImplementedError` for the 3 dict* methods, TestCursor implemented all of them already. No need to maintain deadcode IMO.